### PR TITLE
nsis: init at 3.04

### PR DESCRIPTION
**Pull request by pedropombeiro**
###### Motivation for this change

Add support for [NSIS](https://nsis.sourceforge.io) (Nullsoft Scriptable Install System) in Nix.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by symphorien** [comment]
Have you tried building from source? There is a .travis.yml file at the root of their repo so it might be doable.

**Comment by PombeirP** [comment]
It's most likely doable, but I needed something quick to get started and decided to contribute it. Hopefully someone can improve it later, but I figured it might help other people who need to use this tool to generate Windows installers.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23332,4 +23332,5 @@ in
     stdenv = crossLibcStdenv;
   };
 
+  nsis = callPackage ../development/tools/nsis { };
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix]: You put this in the very end of all-packages. There is an appropriate section:
https://github.com/NixOS/nixpkgs/blob/198b7ea83ad53fbb679768675b23978b0d090c95/pkgs/top-level/all-packages.nix#L8296

Comment (PombeirP) [reply pkgs/top-level/all-packages.nix]: Wasn't aware of that, thanks!

**Comment by PombeirP** [comment]
Is anything missing in this PR before it can be merged?

**Comment by veprbl** [comment]
It's a bit strange that we are repackaging a ubuntu package. Looking at AUR [1] this doesn't seem to be such a complicated package. It does require a cross-compiled zlib though, which nixpkgs should be able to provide.

[1] https://aur.archlinux.org/packages/nsis/

**Comment by veprbl** [comment]
Regarding cross compilation, this might help: https://matthewbauer.us/blog/beginners-guide-to-cross.html

**Comment by PombeirP** [comment]
Yeah, at the time I wrote this PR I was making my first steps in Nix, so did what would be quicker to get the result I needed. Since I saw that there were other packages which were repackaging dpkgs I thought it would be useful to contribute. I agree though that we should have a nsis package built from source, I just don't know if I'll have the time to dedicate to this.

**Comment by veprbl** [comment]
The other reason why this is still not merged might be that we don't have someone to test it. I'm not sure who to cc for this.

**Comment by PombeirP** [comment]
@veprbl I've updated the PR to build from source and used it in https://github.com/status-im/status-react/pull/7506 with success.

**Comment by veprbl** [comment]
@GrahamcOfBorg build nsis

**Review by veprbl — APPROVED** [review]

File: pkgs/development/tools/nsis/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  version = "3.04";
+
+  src =
+    fetchurl {
+      url = "https://vorboss.dl.sourceforge.net/project/nsis/NSIS%203/${version}/nsis-${version}-src.tar.bz2";
+      sha256 = "1xgllk2mk36ll2509hd31mfq6blgncmdzmwxj3ymrwshdh23d5b0";
+    };
+  srcWinDistributable =
+    fetchzip {
+      url = "https://vorboss.dl.sourceforge.net/project/nsis/NSIS%203/${version}/nsis-${version}.zip";
```
Comment (symphorien) [inline pkgs/development/tools/nsis/default.nix]:
```suggestion
      url = "mirror://sourceforge/project/nsis/NSIS%203/${version}/nsis-${version}.zip";
```

**Review by symphorien — APPROVED** [review]

**Comment by veprbl** [comment]
cc @angerman

**Pull request merged**